users/tracing/groups.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Package tracing contains middlewares that will add spans
+// to existing traces.
+package tracing
+
+import (
+	"context"
+
+	"github.com/mainflux/mainflux/users"
+	opentracing "github.com/opentracing/opentracing-go"
+)
+
+const (
+	assignUser         = "assign_user"
+	saveGroupOp        = "save_group_op"
+	deleteGroupOp      = "delete_group_op"
+	updateGroupOp      = "update_group"
+	retrieveGroupByID  = "retrieve_group_by_id"
+	retrieveAll        = "retrieve_all_group"
+	retrieveByName     = "retrieve_by_name"
+	retrieveAllForUser = "retrieve_all_group_for_user"
+	removeUser         = "remove_user_from_group"
+)
+
+var _ users.GroupRepository = (*groupRepositoryMiddleware)(nil)
+
+type groupRepositoryMiddleware struct {
+	tracer opentracing.Tracer
+	repo   users.GroupRepository
+}
+
+// GroupRepositoryMiddleware tracks request and their latency, and adds spans
+// to context.
```

Comment:
Move to upper line - ty to keep all comments in one line if it is not longer than code below.